### PR TITLE
CAMEL-24463: Add KeyValueRepository implementations for Caffeine, Ehcache, JCache, and Hazelcast

### PR DESCRIPTION
**Pull request by gnodet**
_Claude Code on behalf of @gnodet_

## Summary

Adds `KeyValueRepository` implementations for Caffeine, Ehcache, JCache, and Hazelcast.

### `CaffeineKeyValueRepository` (`camel-caffeine`)
- Uses Caffeine's native per-entry `Expiry` API for TTL support
- Atomic CAS via `cache.asMap().computeIfPresent()`
- Configurable `maximumSize` (default 10,000)
- Stores Java objects directly (in-process, no serialization needed)

### `EhcacheKeyValueRepository` (`camel-ehcache`)
- Takes a pre-configured `CacheManager` and cache name
- TTL via `TtlValue` wrapper with lazy eviction on read
- CAS: uses default non-atomic (TtlValue wrapper prevents native CAS)
- Stores Java objects directly (in-process, no serialization needed)

### `JCacheKeyValueRepository` (`camel-jcache`)
- JSR-107 standard cache API — works with any JCache provider
- TTL via `TtlValue` wrapper with lazy eviction
- CAS: uses default non-atomic (TtlValue wrapper prevents native CAS)
- Stores Java objects directly (in-process, no serialization needed)

### `HazelcastKeyValueRepository` (`camel-hazelcast`)
- Uses Hazelcast `IMap` for distributed key-value storage
- Native per-entry TTL via `IMap.put(key, value, ttl, unit)`
- Atomic CAS via `IMap.replace(K,V,V)` + `IMap.remove(K,V)`; distributed lock for TTL case
- Values serialized through `KeyValueRepositoryHelper` and stored as `byte[]` for consistent format across all persistent backends

### Serialization approach

In-process backends (Caffeine, Ehcache, JCache) store Java objects directly — no serialization overhead. Distributed/persistent backends (Hazelcast) serialize through `KeyValueRepositoryHelper` for a consistent format with JDBC, JPA, Cassandra, Kafka, Redis, and Infinispan.

### All implementations
- Full `KeyValueRepository` SPI: get, put (with TTL), delete, contains, keys, clear, putIfAbsent, size
- CAS overrides where the backend supports it
- `@Metadata`, `@Configurer`, `@ManagedResource` annotations

## Test plan
- [x] Hazelcast unit tests: **25/25** ✅
- [x] Caffeine unit tests: **21/21** ✅
- [x] Ehcache unit tests: **21/21** ✅
- [x] JCache unit tests: **21/21** ✅

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: components/camel-hazelcast/src/test/java/org/apache/camel/component/hazelcast/HazelcastKeyValueRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class HazelcastKeyValueRepositoryTest {
+
+    @RegisterExtension
+    static HazelcastService hazelcastService = HazelcastServiceFactory.createSingletonService();
```
Comment (davsclaus) [inline components/camel-hazelcast/src/test/java/org/apache/camel/component/hazelcast/HazelcastKeyValueRepositoryTest.java:43]: Minor: this `@RegisterExtension` field is never used elsewhere in the class — the test creates its own instance directly via `Hazelcast.newHazelcastInstance(null)` a few lines below instead of going through `hazelcastService`. The sibling `HazelcastIdempotentRepositoryTest` at least calls `hazelcastService.createConfiguration(...)` to get a `Config` with multicast/auto-detection disabled, avoiding one test's embedded node accidentally joining another's cluster (though that test then also passes `null` to `newHazelcastInstance` instead of the computed config, so the pattern is already a bit inconsistent in the codebase). Consider either wiring this into `createConfiguration(...)` for test isolation, or dropping the unused field.

**Review by davsclaus — APPROVED** [review]
Reviewed against the project's build/test/formatting/security-annotation conventions (CLAUDE.md), plus git history of the affected files and the broader CAMEL-24463 series (#25631, #25863 merged; #25818, #25993, #26009 open).

This is a well-executed PR. All four implementations correctly implement the `KeyValueRepository` SPI: CAS semantics are atomic where the backend supports it (Caffeine via `ConcurrentMap.computeIfPresent`, Hazelcast via `IMap.replace`/`remove` plus a distributed lock for the TTL-replace case) and honestly fall back to the SPI's documented non-atomic default where the `TtlValue` wrapper prevents native CAS (Ehcache, JCache). Lifecycle handling correctly avoids closing externally-supplied resources the repository doesn't own (Ehcache `CacheManager`, JCache `Cache` via `JCacheManager`'s no-op close when constructed from a cache directly). Hazelcast's serialization allow-list (`HazelcastSerializationFilterHelper.applyDefault`) is applied consistent with the project's security conventions. Tests are AssertJ + Awaitility throughout with no `Thread.sleep`, and cover CRUD, TTL expiration, and CAS paths well.

One minor, non-blocking item left as an inline comment below on `HazelcastKeyValueRepositoryTest`.

Two further non-blocking notes:
- No `.adoc` doc-page updates for the new beans, but this matches existing practice for sibling repositories in these same components (only Ehcache's idempotent/aggregation repos are documented) — not a regression introduced by this PR. Worth considering a single consolidated "KeyValueRepository backends" doc page once the full family (Redis, Infinispan, JDBC, JPA, Cassandra, Kafka from #25818/#25993) has landed.
- #26009 ("Improve KeyValueRepository SPI") is also open against the same interface this PR builds on. If it merges first, this PR will need another rebase — just a sequencing note, not a defect.

This review is a rules-and-conventions check plus manual reading of the diff and git history per the project's `CLAUDE.md` and OSS Helper standards — it does not replace SonarCloud, CodeRabbit, or similar static-analysis/review tools.

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

File: components/camel-hazelcast/src/test/java/org/apache/camel/component/hazelcast/HazelcastKeyValueRepositoryTest.java (hunk 1 above)
Comment (gnodet) [reply components/camel-hazelcast/src/test/java/org/apache/camel/component/hazelcast/HazelcastKeyValueRepositoryTest.java:43]: _Claude Code on behalf of gnodet_

Addressed in a94be8edda4 — the test now uses `hazelcastService.createConfiguration(null, 0, null, MAP_NAME)` and passes the resulting `Config` to `Hazelcast.newHazelcastInstance(config)` instead of `null`, ensuring proper multicast/auto-detection isolation between test instances.
